packages/eslint-plugin/src/rules/block-spacing.ts

```diff
@@ -0,0 +1,166 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_TOKEN_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+
+const baseRule = getESLintCoreRule('block-spacing');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'block-spacing',
+  meta: {
+    type: 'layout',
+    docs: {
+      description:
+        'Disallow or enforce spaces inside of blocks after opening block and before closing block',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: 'whitespace',
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: ['always'],
+
+  create(context, [whenToApplyOption]) {
+    const sourceCode = context.getSourceCode();
+    const baseRules = baseRule.create(context);
+    const always = whenToApplyOption !== 'never';
+    const messageId = always ? 'missing' : 'extra';
+    /**
+     * Gets the open brace token from a given node.
+     * @returns The token of the open brace.
+     */
+    function getOpenBrace(
+      node: TSESTree.TSEnumDeclaration,
+    ): TSESTree.PunctuatorToken {
+      // guaranteed for enums
+      // This is the only change made here from the base rule
+      return sourceCode.getFirstToken(node, {
+        filter: token =>
+          token.type === AST_TOKEN_TYPES.Punctuator && token.value == '{',
+      }) as TSESTree.PunctuatorToken;
```

Comment:
The only significant change from the base rule, it worked flawlessly for interface and object type literal.
Added a filter for punctuator because enums don't have a body node in the AST

https://github.com/eslint/eslint/blob/3ae0574fc78c4a2b406625e4792cb2859cb9bcb1/lib/rules/block-spacing.js#L60